Storage/test.cc

```diff
@@ -48,6 +48,8 @@ namespace transactional_storage_test {
 CURRENT_STRUCT(Element) {
   CURRENT_FIELD(x, int32_t);
   CURRENT_CONSTRUCTOR(Element)(int32_t x = 0) : x(x) {}
+  // Dummy key function to support key-only deleter.
+  int32_t key() const { return x; }
```

Comment:
Shouldn't `Element` already have it own key field/method defined to be used in the storage?

On the general note, why not have some `CURRENT_PRIMARY_KEY(...);` (and `CURRENT_PRIMARY_KEY_METHOD()` macro, much like we have `CURRENT_TIMESTAMP()`?